document-generator-accounts/src/test/java/uk/gov/companieshouse/document/generator/accounts/mapping/smallfull/mappers/ApiToBalanceSheetMapperTest.java

```diff
@@ -0,0 +1,246 @@
+        assertEquals(new Long(VALUE_ONE), calledUpSharedCapitalNotPaid.getPreviousAmount());
+    }
+
+    @Test
+    @DisplayName("tests that the current period values map to called up share capital not paid IXBRL model")
+    void testApiToCalledUpShareCapitalNotPaidMapsCurrentOnly() {
+
+        CalledUpSharedCapitalNotPaid calledUpSharedCapitalNotPaid = ApiToBalanceSheetMapper
+                .INSTANCE.apiToCalledUpSharedCapitalNotPaid(setCurrentPeriod(), null);
+
+        assertNotNull(calledUpSharedCapitalNotPaid);
+        assertEquals(new Long(VALUE_ONE), calledUpSharedCapitalNotPaid.getCurrentAmount());
+    }
+
+    private CurrentPeriodApi setCurrentPeriod() {
+
+        CurrentPeriodApi currentPeriod = new CurrentPeriodApi();
+
+        currentPeriod.setBalanceSheetApi(setBalanceSheetValues());
+
+        return currentPeriod;
+    }
+
+    private PreviousPeriodApi setPreviousPeriod() {
+
+        PreviousPeriodApi previousPeriod = new PreviousPeriodApi();
+
+        previousPeriod.setBalanceSheet(setBalanceSheetValues());
+
+        return previousPeriod;
+    }
+
+    private BalanceSheetApi setBalanceSheetValues() {
+
+        BalanceSheetApi balanceSheet = new BalanceSheetApi();
+
+        CapitalAndReservesApi capitalAndReserves = new CapitalAndReservesApi();
+        capitalAndReserves.setCalledUpShareCapital(new Long(VALUE_ONE));
+        capitalAndReserves.setOtherReserves(new Long(VALUE_TWO));
+        capitalAndReserves.setProfitAndLoss(new Long(VALUE_THREE));
+        capitalAndReserves.setSharePremiumAccount(new Long(VALUE_ONE));
+        capitalAndReserves.setTotalShareholdersFund(new Long(VALUE_TWO));
+
+        CurrentAssetsApi currentAssets = new CurrentAssetsApi();
+        currentAssets.setCashInBankAndInHand(new Long(VALUE_ONE));
+        currentAssets.setDebtors(new Long(VALUE_TWO));
+        currentAssets.setStocks(new Long(VALUE_THREE));
+        currentAssets.setTotal(new Long(VALUE_ONE));
+
+        FixedAssetsApi fixedAssets = new FixedAssetsApi();
+        fixedAssets.setTangible(new Long(VALUE_ONE));
+        fixedAssets.setTotal(new Long(VALUE_TWO));
+
+        OtherLiabilitiesOrAssetsApi otherLiabilitiesOrAssets = new OtherLiabilitiesOrAssetsApi();
+        otherLiabilitiesOrAssets.setTotalNetAssets(new Long(VALUE_ONE));
+        otherLiabilitiesOrAssets.setAccrualsAndDeferredIncome(new Long(VALUE_TWO));
+        otherLiabilitiesOrAssets.setCreditorsAfterOneYear(new Long(VALUE_THREE));
+        otherLiabilitiesOrAssets.setCreditorsDueWithinOneYear(new Long(VALUE_ONE));
+        otherLiabilitiesOrAssets.setNetCurrentAssets(new Long(VALUE_TWO));
+        otherLiabilitiesOrAssets.setPrepaymentsAndAccruedIncome(new Long(VALUE_THREE));
+        otherLiabilitiesOrAssets.setTotalAssetsLessCurrentLiabilities(new Long(VALUE_ONE));
+        otherLiabilitiesOrAssets.setProvisionForLiabilities(new Long(VALUE_TWO));
+
+        balanceSheet.setCapitalAndReservesApi(capitalAndReserves);
```

Comment:
I would move the instantiations to different methods as this `setBalanceSheetValues` is going to keep growing when adding new notes. 

```
balanceSheet.setCapitalAndReservesApi(createCapitalAndReserves());
balanceSheet.setCurrentAssetsApi(createCurrentAssets); 
...
...
```

```
private CapitalAndReservesApi createCapitalAndReserves() {
    CapitalAndReservesApi capitalAndReserves = new CapitalAndReservesApi();
    capitalAndReserves.setCalledUpShareCapital(new Long(VALUE_ONE));
    capitalAndReserves.setOtherReserves(new Long(VALUE_TWO));
    capitalAndReserves.setProfitAndLoss(new Long(VALUE_THREE));
    capitalAndReserves.setSharePremiumAccount(new Long(VALUE_ONE));
    capitalAndReserves.setTotalShareholdersFund(new Long(VALUE_TWO));
        
    return capitalAndReserves;
}
```

Reply:
can do, although balance sheet shouldn't grow anymore. Notes will be separate.

Reply:
done